src/dbos.ts

```diff
@@ -908,7 +908,9 @@ export class DBOS {
           timerFuncID,
         );
         if (!rres) return null;
-        if (rres?.cancelled) throw new DBOSTargetWorkflowCancelledError(workflowID); // TODO: Make semantically meaningful
+        if (rres?.cancelled) {
+          throw new DBOSAwaitedWorkflowCancelledError(DBOS.workflowID ? DBOS.workflowID : workflowID);
```

Comment:
if we don't do that, it is possible for a parent handle get result to call a child handle get result and get back this error about the wrong workflow

Reply:
The message looks ambiguous: `DBOS.workflowID` is the current workflow, while `workflowID` is the target.